docs/features/mcp/README.md

```diff
@@ -0,0 +1,324 @@
+
+### HTTPS Enforcement
+
+MCP requests require HTTPS by default. For local development, you can disable this requirement:
+
+```php
+add_filter( 'woocommerce_mcp_allow_insecure_transport', '__return_true' );
+```
+
+### Permission Validation
+
+The transport layer validates operations against API key permissions:
+
+- `read` permissions: Allow GET requests
+- `write` permissions: Allow POST, PUT, PATCH, DELETE requests
+- `read_write` permissions: Allow all operations
+
+## Server Endpoint
+
+The WooCommerce MCP server is available at:
+
+```text
+https://yourstore.com/wp-json/woocommerce/mcp
+```
+
+## Connecting to the MCP Server
+
+### Proxy Architecture
+
+The current MCP implementation uses a **local proxy approach** to connect MCP clients with WordPress servers:
+
+- **MCP Clients** (like Claude Code) communicate using the MCP protocol over stdio/JSON-RPC
+- **Local Proxy** (`@automattic/mcp-wordpress-remote`) runs on your machine and translates MCP protocol messages to HTTP requests
+- **WordPress MCP Server** receives HTTP requests and processes them through the WordPress Abilities system
+
+This proxy pattern is commonly used in MCP integrations to bridge protocol differences and handle authentication. The `mcp-wordpress-remote` package acts as a protocol translator, converting the stdio-based MCP communication that clients expect into the HTTP REST API calls that WordPress understands.
+
+**Future Evolution**: While this proxy approach provides a robust foundation, future implementations may explore direct MCP protocol support within WordPress or alternative connection methods as the MCP ecosystem evolves.
+
+### Claude Code Integration
+
+To connect Claude Code to your WooCommerce MCP server:
+
+1. Go to **WooCommerce → Settings → Advanced → REST API**
+2. Create a new API key with "Read/Write" permissions
+3. Configure MCP with your API key using Claude Code:
+
+```bash
+claude mcp add woocommerce_mcp \
+  --env WP_API_URL=https://yourstore.com/wp-json/woocommerce/mcp \
+  --env CUSTOM_HEADERS='{"X-MCP-API-Key": "YOUR_CONSUMER_KEY:YOUR_CONSUMER_SECRET"}' \
+  -- npx -y @automattic/mcp-wordpress-remote@latest
+```
+
+### Manual MCP Client Configuration
+
+For other MCP clients, add this configuration to your MCP settings. This configuration tells the MCP client to run the `mcp-wordpress-remote` proxy locally, which will handle the communication with your WordPress server:
+
+```json
+{
+  "mcpServers": {
+    "woocommerce_mcp": {
+      "type": "stdio",
+      "command": "npx",
+      "args": [
+        "-y",
+        "@automattic/mcp-wordpress-remote@latest"
+      ],
+      "env": {
+        "WP_API_URL": "https://yourstore.com/wp-json/woocommerce/mcp",
+        "CUSTOM_HEADERS": "{\"X-MCP-API-Key\": \"YOUR_CONSUMER_KEY:YOUR_CONSUMER_SECRET\"}"
+      }
+    }
+  }
+}
+```
+
+**Important**: Replace `YOUR_CONSUMER_KEY:YOUR_CONSUMER_SECRET` with your actual WooCommerce API credentials.
```

Comment:
One of the more common issues I've seen is with the user's default `npx` command being the wrong version, or SSL issues in local environments. It could be helpful to include a link directly to the troubleshooting docs for `mcp-wordpress-remote` for developers who are new to it: https://github.com/Automattic/mcp-wordpress-remote/blob/trunk/Docs/troubleshooting.md